src/main/java/io/confluent/idesidecar/restapi/proxy/CCloudApiProcessor.java

```diff
@@ -0,0 +1,53 @@
+package io.confluent.idesidecar.restapi.proxy;
+
+import static io.vertx.core.http.HttpHeaders.AUTHORIZATION;
+
+import io.confluent.idesidecar.restapi.connections.CCloudConnectionState;
+import io.confluent.idesidecar.restapi.exceptions.ProcessorFailedException;
+import io.confluent.idesidecar.restapi.processors.Processor;
+import io.confluent.idesidecar.restapi.util.UriUtil;
+import io.vertx.core.Future;
+import io.vertx.core.MultiMap;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+
+@ApplicationScoped
+public class CCloudApiProcessor extends Processor<ProxyContext, Future<ProxyContext>> {
+
+  @Inject
+  UriUtil uriUtil;
+
+  @Override
+  public Future<ProxyContext> process(ProxyContext context) {
+    String clusterId = context.getRequestHeaders().get("X-Cluster-Id");
+    var connectionState = context.getConnectionState();
+    if (connectionState instanceof CCloudConnectionState cCloudConnection
+        && (clusterId == null
+        || clusterId.isEmpty())) {
```

Comment:
Do you want to keep this check? If so, maybe consider providing a more specific error message in case the user passed the `X-Cluster-Id` header, like: `This route does not support the request header 'X-Cluster-Id'.`

Reply:
good point. I moved to check to later in the if/else chain and added an error message like the one you suggested